client/header/activity-panel/panels/inbox/card.js

```diff
@@ -39,39 +49,182 @@ class InboxNoteCard extends Component {
+				<NoteAction
+					key={ note.id }
+					noteId={ note.id }
+					label={ __( "Yes, I'm sure", 'woocommerce-admin' ) }
+					actionCallback={ this.closeDismissModal }
+					dismiss={ true }
+					dismissType={ dismissType }
+				/>
+			);
+		};
+		return (
+			<Modal
+				title={
+					<Fragment>
+						{ __( 'Are you sure?', 'woocommerce-admin' ) }
+					</Fragment>
+				}
+				onRequestClose={ () => this.closeDismissModal() }
+				className="woocommerce-inbox-dismiss-confirmation_modal"
+			>
+				<div className="woocommerce-inbox-dismiss-confirmation_wrapper">
+					<p>
+						{ __(
+							'Dismissed messages cannot be viewed again',
+							'woocommerce-admin'
+						) }
+					</p>
+					<div className="woocommerce-inbox-dismiss-confirmation_buttons">
+						<Button
+							isDefault
+							onClick={ () => this.closeDismissModal() }
+						>
+							{ __( 'Cancel', 'woocommerce-admin' ) }
+						</Button>
+						{ getDismissButtonFromActions() }
+					</div>
+				</div>
+			</Modal>
+		);
+	}
+
 	render() {
 		const { lastRead, note } = this.props;
+		const { isDismissModalOpen } = this.state;
 
 		const getButtonsFromActions = () => {
 			if ( ! note.actions ) {
 				return [];
 			}
 			return note.actions.map( ( action ) => (
-				<NoteAction key={ note.id } noteId={ note.id } action={ action } />
+				<NoteAction
+					key={ note.id }
+					noteId={ note.id }
+					action={ action }
+				/>
 			) );
 		};
 
+		const unread =
+			! lastRead ||
+			! note.date_created_gmt ||
+			new Date( note.date_created_gmt + 'Z' ).getTime() > lastRead;
+		const actions = getButtonsFromActions( note );
+		const actionsList = Array.isArray( actions ) ? actions : [ actions ];
+		const date = note.date_created;
+		const hasImage = note.layout !== 'plain';
+		const cardClassName = classnames(
+			'woocommerce-inbox-message',
+			note.layout,
+			{
+				'message-is-unread': unread,
+			}
+		);
+
 		return (
 			<VisibilitySensor onChange={ this.onVisible }>
-				<ActivityCard
-					className={ classnames( 'woocommerce-inbox-activity-card', {
-						actioned: note.status !== 'unactioned',
-					} ) }
-					title={ note.title }
-					date={ note.date_created }
-					icon={ <Gridicon icon={ note.icon } size={ 48 } /> }
-					unread={
-						! lastRead ||
-						! note.date_created_gmt ||
-						new Date( note.date_created_gmt + 'Z' ).getTime() >
-							lastRead
-					}
-					actions={ getButtonsFromActions( note ) }
-				>
-					<span
-						dangerouslySetInnerHTML={ sanitizeHTML( note.content ) }
-					/>
-				</ActivityCard>
+				<section className={ cardClassName }>
+					{ hasImage && (
+						<div className="woocommerce-inbox-message__image">
+							<img src={ note.image } alt="" />
+						</div>
+					) }
+					<div className="woocommerce-inbox-message__wrapper">
+						<div className="woocommerce-inbox-message__content">
+							{ date && (
+								<span className="woocommerce-inbox-message__date">
+									{ moment.utc( date ).fromNow() }
```

Comment:
🤔 my memory is foggy here, but is moment localization handled upstream by WP Core? 

I think I answered this myself via https://github.com/woocommerce/woocommerce-admin/blob/master/webpack.config.js#L51 but leaving this here just in case. But @octaedro curious if you tested this in a different locale than English yet.

Reply:
Thank you @timmyc for your comment. 

My approach here was trying to decouple the inbox notifications code from the activity cards. I essentially moved this file from [/client/header/activity-panel/activity-card/index.js](https://github.com/woocommerce/woocommerce-admin/blob/master/client/header/activity-panel/activity-card/index.js) to the `inbox` folder.
I thought that by doing this it will be easier to add the inbox messages in the new dashboard later.

I didn't test it with a different locale than English, this is a good suggestion 🙂
Thanks!